packages/open-next/src/core/routing/util.ts

```diff
@@ -105,14 +105,15 @@ export function convertRes(res: OpenNextNodeResponse): InternalResult {
   const isBase64Encoded =
     isBinaryContentType(headers["content-type"]) ||
     !!headers["content-encoding"];
+  let index = 0;
   const body = new ReadableStream({
     pull(controller) {
-      if (!res._chunks || res._chunks.length === 0) {
+      if (!res._chunks || index >= res._chunks.length) {
         controller.close();
         return;
       }
 
-      controller.enqueue(res._chunks.shift());
```

Comment:
The intent here was to allow freeing memory early.

What's the problem with that? Having to move memory on `shift`s? it should be only refs so not too much?

Reply:
This constantly triggers GC and when streaming this would create unnecessary GC pressure, which would impact the performance. Since we already allocated it, there is no benefit of an additional GC pressure